charts/kube-prometheus-stack/values.yaml

```diff
@@ -1013,6 +1013,22 @@ grafana:
   ##
   forceDeployDatasources: false
 
+  ## ConfigMapDatasources Create datasources as configmaps
+  ##
+  configMapDatasources: true
+
+  ## OperatorDatasources Create datasources as crds
+  ##
+  
+  operatorDatasources: false
+
+  ## OperatorInstanceSelector Selects the instance of the operator to use for grafana operator
+  ##
+
+  operatorInstanceSelector: 
+    app.kubernetes.io/instance: grafana
+    app.kubernetes.io/name: grafana-operator
```

Comment:
There is a known helm limitation where keys can't be deleted, if kube-prometheus-stack is part of a sub-chart.

If end-users do something like

```
operatorInstanceSelector: 
    app.kubernetes.io/instance: ~
    app.kubernetes.io/name: another-grafana-operator
    instance: grafana
```

`app.kubernetes.io/instance` would still remains in addition of `instance`. Based on the limitation, my recommendation would be to leave `operatorInstanceSelector` to an empty map.